### PR TITLE
add repr to RuntimeValue to fix #120

### LINKED ISSUE #120: PyObject value in Result can cause JepException
```
19/03/06 17:55:06 ERROR SocketSession: IllegalStateException: jep.JepException: Invalid thread access.
polynote.kernel.RuntimeError: IllegalStateException: jep.JepException: Invalid thread access.
	at polynote.kernel.ErrorResult$.apply(Result.scala:136)
	at polynote.server.SocketSession$$anonfun$toResponse$1$$anonfun$4.apply(SocketSession.scala:89)
	at polynote.server.SocketSession$$anonfun$toResponse$1$$anonfun$4.apply(SocketSession.scala:88)
	at fs2.Stream$$anonfun$handleErrorWith$extension$1.apply(Stream.scala:1324)
	at fs2.Stream$$anonfun$handleErrorWith$extension$1.apply(Stream.scala:1324)
	at fs2.internal.FreeC$$anonfun$handleErrorWith$1.apply(FreeC.scala:54)
	at fs2.internal.FreeC$$anonfun$handleErrorWith$1.apply(FreeC.scala:51)
	at fs2.internal.FreeC$ViewL$.mk(FreeC.scala:214)
	at fs2.internal.FreeC$ViewL$.apply(FreeC.scala:206)
	at fs2.internal.FreeC.viewL(FreeC.scala:67)
	at fs2.internal.Algebra$.fs2$internal$Algebra$$go$1(Algebra.scala:220)
	at fs2.internal.Algebra$$anonfun$fs2$internal$Algebra$$closeAndGo$1$1$$anonfun$apply$10.apply(Algebra.scala:319)
	at fs2.internal.Algebra$$anonfun$fs2$internal$Algebra$$closeAndGo$1$1$$anonfun$apply$10.apply(Algebra.scala:317)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:139)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:351)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:372)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:312)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:136)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:351)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:372)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:312)
	at cats.effect.internals.Callback$AsyncIdempotentCallback.run(Callback.scala:127)
	at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:70)
	at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:36)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.cats$effect$internals$TrampolineEC$JVMTrampoline$$super$startLoop(TrampolineEC.scala:93)
	at cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply$mcV$sp(TrampolineEC.scala:93)
	at cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply(TrampolineEC.scala:93)
	at cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply(TrampolineEC.scala:93)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:92)
	at cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
	at cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:44)
	at cats.effect.internals.Callback$AsyncIdempotentCallback.apply(Callback.scala:133)
	at cats.effect.internals.Callback$AsyncIdempotentCallback.apply(Callback.scala:120)
	at cats.effect.concurrent.Deferred$ConcurrentDeferred$$anonfun$2.apply(Deferred.scala:205)
	at cats.effect.concurrent.Deferred$ConcurrentDeferred$$anonfun$2.apply(Deferred.scala:205)
	at cats.effect.concurrent.Deferred$ConcurrentDeferred$$anonfun$1.apply$mcV$sp(Deferred.scala:241)
	at cats.effect.concurrent.Deferred$ConcurrentDeferred$$anonfun$1.apply(Deferred.scala:241)
	at cats.effect.concurrent.Deferred$ConcurrentDeferred$$anonfun$1.apply(Deferred.scala:241)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:87)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:351)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:372)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:312)
	at cats.effect.internals.IOShift$Tick.run(IOShift.scala:36)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.IllegalStateException: jep.JepException: Invalid thread access.
	at jep.python.PyObject.checkValidRuntime(PyObject.java:112)
	at jep.python.PyObject.toString(PyObject.java:679)
	at polynote.kernel.util.KernelContext$$anonfun$5.apply(KernelContext.scala:85)
	at polynote.kernel.util.KernelContext$$anonfun$5.apply(KernelContext.scala:85)
	at scala.Option.map(Option.scala:146)
	at polynote.kernel.util.KernelContext.reprsOf(KernelContext.scala:85)
	at polynote.kernel.ResultValue$.apply(Result.scala:175)
	at polynote.kernel.util.RuntimeSymbolTable$RuntimeValue.toResultValue(RuntimeSymbolTable.scala:112)
	at polynote.kernel.PolyKernel$$anonfun$currentSymbols$1$$anonfun$apply$38.apply(PolyKernel.scala:204)
	at polynote.kernel.PolyKernel$$anonfun$currentSymbols$1$$anonfun$apply$38.apply(PolyKernel.scala:204)
	at scala.collection.immutable.List.map(List.scala:288)
	at polynote.kernel.PolyKernel$$anonfun$currentSymbols$1.apply(PolyKernel.scala:204)
	at polynote.kernel.PolyKernel$$anonfun$currentSymbols$1.apply(PolyKernel.scala:204)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:87)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:351)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:372)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:312)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:136)
	at cats.effect.internals.IORunLoop$.startCancelable(IORunLoop.scala:41)
	at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:86)
	at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:70)
	at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:36)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.cats$effect$internals$TrampolineEC$JVMTrampoline$$super$startLoop(TrampolineEC.scala:93)
	at cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply$mcV$sp(TrampolineEC.scala:93)
	at cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply(TrampolineEC.scala:93)
	at cats.effect.internals.TrampolineEC$JVMTrampoline$$anonfun$startLoop$1.apply(TrampolineEC.scala:93)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:92)
	at cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
	at cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:44)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:72)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:52)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:136)
	at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:34)
	at cats.effect.internals.IOBracket$$anonfun$apply$1.apply(IOBracket.scala:44)
	at cats.effect.internals.IOBracket$$anonfun$apply$1.apply(IOBracket.scala:34)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:337)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:119)
	at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:34)
	at cats.effect.internals.IOBracket$$anonfun$apply$1.apply(IOBracket.scala:44)
	at cats.effect.internals.IOBracket$$anonfun$apply$1.apply(IOBracket.scala:34)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:337)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:119)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:351)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:372)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:312)
	at cats.effect.internals.IOShift$Tick.run(IOShift.scala:36)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
	... 2 more
Caused by: jep.JepException: Invalid thread access.
	at jep.Jep.isValidThread(Jep.java:319)
	at jep.python.PyObject.checkValid(PyObject.java:94)
	at jep.python.PyObject.checkValidRuntime(PyObject.java:110)
	... 52 more
```

### PR DESCRIPTION
**Pull request by indig**
@jeremyrsmith, I tried using `IO` per your comment in #120 but wasn't able to get it to work nicely, so I opted for this approach which might be a little clunkier but hopefully RST will go away 😄

### CODE REVIEW

File: polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala [hunk 1]
```diff
@@ -109,12 +112,12 @@ final class RuntimeSymbolTable(
...
     override def hashCode(): Int = name.hashCode()
 
-    def toResultValue: ResultValue = ResultValue(kernelContext, name, scalaTypeHolder, value, sourceCellId)
+    def toResultValue: ResultValue = ResultValue(kernelContext, name, reprs, scalaTypeHolder, value, sourceCellId)
   }
 
   object RuntimeValue {
     def apply(name: String, value: Any, source: Option[LanguageInterpreter[IO]], sourceCell: CellID): RuntimeValue = RuntimeValue(
-      name, value, kernelContext.inferType(value), source, sourceCell
+      name, value, kernelContext.reprsOf(value, kernelContext.inferType(value)), kernelContext.inferType(value), source, sourceCell
```
Comment (jeremyrsmith) [inline polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala]: Can we `inferType` only once here? It's actually not that cheap to do :(

File: polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala [hunk 2]
```diff
@@ -29,7 +30,7 @@ final class RuntimeSymbolTable(
 
   private val newSymbols: Topic[IO, RuntimeValue] =
     Topic[IO, RuntimeValue]{
-      val kernel = RuntimeValue("kernel", polynote.runtime.Runtime, global.typeOf[polynote.runtime.Runtime.type], None, -1)
+      val kernel = RuntimeValue("kernel", polynote.runtime.Runtime, None, -1)
```
Comment (jeremyrsmith) [inline polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala]: hmm, won't this cause an `inferType` for `kernel`? Maybe keep the `global.typeOf` and just manually create a `StringRepr` to pass in (even empty string repr would be fine for the kernel object I think)

File: polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala (hunk 1 above)
Comment (indig) [reply polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala]: Thanks for the catch, done!

File: polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala (hunk 2 above)
Comment (indig) [reply polynote-kernel/src/main/scala/polynote/kernel/util/RuntimeSymbolTable.scala]: Done! For some reason IntelliJ gives me angry red underlines when I do that but it compiles fine... 🤷‍♂️

**Pull request merged**